compass/landice/tests/ismip6_forcing/ocean_thermal/process_thermal_forcing.py

```diff
@@ -0,0 +1,314 @@
+                                           output_file, mali_mesh_name,
+                                           mali_mesh_file, method_remap):
+        """
+        Remap the input ismip6 thermal forcing data onto mali mesh
+
+        Parameters
+        ----------
+        input_file: str
+            ismip6 thermal forcing data on its native polarstereo 8km grid
+        output_file : str
+            ismip6 data remapped on mali mesh
+        mali_mesh_name : str
+            name of the mali mesh used to name mapping files
+        mali_mesh_file : str, optional
+            The MALI mesh file if mapping file does not exist
+        method_remap : str, optional
+            Remapping method used in building a mapping file
+        """
+
+        # check if a mapfile
+        mapping_file = f"map_ismip6_8km_to_{mali_mesh_name}_{method_remap}.nc"
+
+        if not os.path.exists(mapping_file):
+            # build a mapping file if it doesn't already exist
+            build_mapping_file(self.config, self.cores, self.logger,
+                               input_file, mapping_file, mali_mesh_file,
+                               method_remap)
+        else:
+            print("Mapping file exists. Remapping the input data...")
+
+        # remap the input data
+        args = ["ncremap",
+                "-i", input_file,
+                "-o", output_file,
+                "-m", mapping_file]
+
+        subprocess.check_call(args)
+
+    def rename_ismip6thermalforcing_to_mali_vars(self, remapped_file_temp,
+                                                 output_file):
+        """
+        Rename variables in the remapped ismip6 input data
+        to the ones that MALI uses.
+
+        Parameters
+        ----------
+        remapped_file_temp : str
+            temporary ismip6 data remapped on mali mesh
+        output_file : str
+            remapped ismip6 data renamed on mali mesh
+        """
+
+        config = self.config
+        section = config['ismip6_ais_ocean_thermal']
+        process_obs_data = section.getboolean('process_obs_data')
+
+        # open dataset in 20 years chunk
+        ds = xr.open_dataset(remapped_file_temp, chunks=dict(time=20),
+                             engine="netcdf4")
+
+        ds["ismip6shelfMelt_zOcean"] = ds.z
+        ds = ds.drop_vars('z')  # dropping 'z' while it's still called 'z'
+
+        # build dictionary for ismip6 variables that MALI takes in
+        if process_obs_data:
+            ismip6_to_mali_dims = dict(
+                z="nISMIP6OceanLayers",
+                ncol="nCells")
+            ds["xtime"] = ("Time", ["2015-01-01_00:00:00".ljust(64)])
+            ds["xtime"] = ds.xtime.astype('S')
+            ds["thermal_forcing"] = ds["thermal_forcing"].\
+                                    expand_dims(dim="Time", axis=0)
+            ds = ds.rename(ismip6_to_mali_dims)
+        else:
+            ismip6_to_mali_dims = dict(
```

Comment:
I'm not sure which lines are responsible for this, but the ismip6shelfMelt_3dThermalForcing field is written out with dimensions in the wrong order: ismip6shelfMelt_3dThermalForcing(Time, nISMIP6OceanLayers, nCells) instead of ismip6shelfMelt_3dThermalForcing(Time, nCells, nISMIP6OceanLayers). This does not cause an error during run time, but leads to ismip6shelfMelt_3dThermalForcing being set to the default value of 2°C everywhere. I only noticed because on the Thwaites mesh all the values of floatingBasalMassBalApplied being written out were identical. I re-ordered them using ncdpq --rdr=Time,nCells,nISMIP6OceanLayers <file.nc> <file.nc>, and now a spatially varying basal mass balance field is being written out.